src/components/interaction-designer/block-editors/GenericContactPropertyEditor.vue

```diff
@@ -195,13 +215,32 @@ export class GenericContactPropertyEditor extends mixins(Lang) {
     })
   }
 
+  get availableContactPropertyFields(): IContactPropertyOption[] {
+    return this.ui.subscriberPropertyFields
+  }
+
+  get flowSelectedContactPropertyField(): IContactPropertyOption | null {
+    const selectedOption = find(
+      this.availableContactPropertyFields,
+      (option: IContactPropertyOption) => option.name === this.block.config.set_contact_property?.property_key,
+    )
+
+    return selectedOption ?? null
+  }
+
+  set flowSelectedContactPropertyField(option: IContactPropertyOption | null) {
+    this.updatePropertyKey(option?.name as string)
+  }
+
   @flowVuexNamespace.Mutation block_updateConfigByPath!: (
     {blockId, path, value}: { blockId: string, path: string, value: string | object }
   ) => void
   @flowVuexNamespace.Mutation block_updateVendorMetadataByPath!: (
     {blockId, path, value}: { blockId: string, path: string, value: string }
   ) => void
   @flowVuexNamespace.Mutation block_removeConfigByKey!: ({blockId, key}: { blockId: string, key: string}) => void
+
+  @State(({trees: {ui}}) => ui) ui!: any
```

Comment:
Generally not a fan of this pattern. I would prefer using the subscriber property fields getter already in `src/store/trees/trees.js`

Reply:
Good point @jcbashdown , I think we already have that `subscriberPropertyFields` as getter there. I just updated the current component. Thanks 🙏